### PR TITLE
Add deploy docs workflow

### PR DESCRIPTION
**Pull request by SooluThomas**
Trying to get the docs hosted using gh-pages

### CODE REVIEW

File: .github/workflows/deploy_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Deploy mkdocs
+
+on:
+  push:
+    branches:
+      - main
+      - soo-deploy-docs
```
Comment (JuLieAlgebra) [inline .github/workflows/deploy_docs.yaml:7]: @SooluThomas Would you like to keep this? I removed the "run on PR" conditions you said to remove before merge, which I agree with, but wasn't sure about this condition! I can see a good argument either way to keep runs on `soo-deploy-docs`. 

Another nice thing though is that we can trigger a new gh deployment by running `pixi run mkdocs gh-deploy` locally if needed: https://www.mkdocs.org/user-guide/deploying-your-docs/

File: .github/workflows/deploy_docs.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install Pixi
+        uses: prefix-dev/setup-pixi@v0.4.3
+        with:
+          pixi-version: ${{ env.PIXI_version }}
```
[the remark this answers is not in the document]
Comment (JuLieAlgebra) [reply .github/workflows/deploy_docs.yaml:33]: Just added the version number!

File: .github/workflows/deploy_docs.yaml (hunk 1 above)
Comment (SooluThomas) [reply .github/workflows/deploy_docs.yaml]: Fixed in https://github.com/NOAA-HES-Capstone/icedyno/pull/5/commits/9d643b50134f606e109c5c54c83a0978091d7b16

File: .github/workflows/deploy_docs.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (SooluThomas) [reply .github/workflows/deploy_docs.yaml:32]: Thank you!

**Comment by SooluThomas** [comment]
Perfect! Thank you so much @JuLieAlgebra for cleaning that up <3

**Review by JuLieAlgebra — APPROVED** [review]

**Review by mthanos7 — APPROVED** [review]

**Pull request merged**